qutip/core/metrics.py

```diff
@@ -108,6 +108,32 @@ def fidelity(A, B):
     return float(np.real(np.sqrt(eig_vals[eig_vals > 0]).sum()))
 
 
+def _hilbert_space_dims(oper):
```

Comment:
Should this function be moved to `core.dimensions`?

Reply:
I think this could be useful in `core.dimensions`, but perhaps let's leave it here for this PR and then move it later. There are some additional considerations in moving it into `core.dimensions` -- e.g. when else will it be used, what is needed to make it return appropriate results for all possible qobjs (e.g. what should happen for bras and kets).